oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/PathUtils.java

```diff
@@ -367,11 +367,13 @@ public static boolean isAncestor(String ancestor, String path) {
         if (denotesRoot(ancestor)) {
             if (denotesRoot(path)) {
                 return false;
+            } else {
+                return path.startsWith(ancestor);
```

Comment:
Wouldn't that make "/a" and ancestor of "/aa"? Maybe we need more unit tests here?

Reply:
There is a test for that case:
https://github.com/apache/jackrabbit-oak/blob/4354285d4a69efc0242a9332064b4c3f91197226/oak-commons/src/test/java/org/apache/jackrabbit/oak/commons/PathUtilsTest.java#L233

The line 371 is inside a block that is executed only when `ancestor = "/"`. So this test is in fact just checking that path starts with "/". This is the same test that is done in the original test in this case.

Reply:
The method has no detailed spec, for instance, it does not say what should be the result with relative paths. And it seems to assume that paths are given in the format "/a/b/c", not as "/a/b/c/". I tried to preserve the semantics of the method.

Reply:
ah, ok.

Reply:
Unless there's test coverage for relative paths, I'd assume that the function is not supposed to be used with relative paths. We should document that, and reject relative paths.

I'm also not that happy with the use of "assert", but that's a separate issue.

Maybe we need to open a ticket to cover the relative path case.

Reply:
I agree that the use of assert in this class is confusing, probably a misguided attempt at reducing runtime overhead. Maybe it made sense 10 years ago, but now I don't think it is justifiable. It should probably be replaced with standard error checking using Preconditions or something similar.